### PR TITLE
Feature/forms

### PR DESCRIPTION
**Pull request by leonardodiasb**
In this branch I:
- Implemented links to new view files which can create new posts and new comments.
- Implemented like button functionality to posts.

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -1,5 +1,35 @@
...
+    @post = Post.find(params[:id])
+    @comments = @post.return_recent_comments.each
+  end
+
+  def new
+    @post = Post.new
+  end
+
+  def create
+    @post = Post.new
```
Comment (GzimAsani) [inline app/controllers/posts_controller.rb]: - [ ] I would suggest you use this way in creating a post `  @post = @current_user.posts.new()`

Please remember [Strong parameters](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/lessons/Controllers_lesson.md#:~:text=Pay%20special%20attention%20to%20the%20strong%20parameters%20description%20in%20Rails%20Guides)

File: app/controllers/comments_controller.rb [hunk 2]
```diff
@@ -0,0 +1,20 @@
+class CommentsController < ApplicationController
+  def new
+    @comment = Comment.new
+  end
+
+  def create
+    @comment = Comment.new
```
Comment (GzimAsani) [inline app/controllers/comments_controller.rb]: - [ ] I would suggest that you should create the comment in this way ``@comment = @current_user.comments.new``

File: app/controllers/likes_controller.rb [hunk 3]
```diff
@@ -0,0 +1,16 @@
+class LikesController < ApplicationController
+  def create
+    @like = Like.new
+    @like.author_id = ApplicationController.current_user.id
```
Comment (GzimAsani) [inline app/controllers/likes_controller.rb]: - [ ] The like controller for create should work like this:
```ruby
 @post = Post.find(params[:post_id])
      @like = @current_user.likes.new(post_id: @post.id)
```

File: app/views/users/show.html.erb [hunk 4]
```diff
@@ -1,2 +1,29 @@
...
+  </div>
+</div>
+<div class="user_bio_container">
+  <p class="bio">Bio</p>
+  <p class="user_bio"><%= @user.bio %></p>
+</div>
+<div class='add_post_box'>
+  <%= link_to 'Add Post', user_new_post_path(@user), class: 'add_post_btn' %>
+</div>
+<% @posts.each do |post| %>
```
Comment (GzimAsani) [inline app/views/users/show.html.erb:17]: - [ ]  I get  this error when I first try to render your project:
Screenshot from 2021-11-06 19-37-59

**Review by GzimAsani — CHANGES_REQUESTED** [review]
# Hi @leonardodiasb                            
Good job so far! Your code looks fine, good job :+1:  :+1: 
There are some issues that you still need to work on to go to the next project but you are almost there!

## To highlight:
- Good job with the gitflow :+1:
- Good PR description :clap: 

## Required Changes ♻️ 
- [ ] _Check inline comments for more information :arrow_double_down:_ 


## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by sinansevgi — APPROVED** [review]
# Approved :heavy_check_mark: 

Hi @leonardodiasb , 

Great work on making the changes requested by the previous code reviewer.
Your project is complete!  
Congratulations! 🎉

You can merge this pull request and continue with your journey :wave: 


Happy Hacking! :computer: 

-----

You can reply to this conversation if you have any questions or need elaboration. :wave: 
Sinan Sevgi


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**